WowCharComparerWebApp/Controllers/User/RegisterController.cs

```diff
@@ -45,14 +56,15 @@ public IActionResult PerformUserRegister(string accountName, string userPassword
                             IsOnline = false,
                             LastLoginDate = DateTime.MinValue,
                             RegistrationDate = DateTime.Now,
-                            Verified = false
+                            Verified = false,
+                            VerificationToken = Guid.NewGuid()
                         };
 
-                        SendVerificationMail(userEmail);
-
                         db.Users.Add(user);
                         db.SaveChanges();
                         transaction.Commit();
+
+                        SendVerificationMail(userEmail, user.VerificationToken);
```

Comment:
Verification email can be send after closing connection with database

Reply:
Fixed